core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -759,6 +758,68 @@ protected RexNode removeCorrelationExpr(
     }
 
     RelNode newRel = relBuilder.build();
+    // Special case where the group by is static (i.e., a select clause with aggregation functions
+    // but without group by).
+    if (rel.getGroupType() == Aggregate.Group.SIMPLE
+        && rel.getGroupSet().isEmpty()
+        && !frame.corDefOutputs.isEmpty()) {
+      final Pair<CorrelationId, Frame> outerFramePair = requireNonNull(this.frameStack.peek());
+      final Frame outFrame = outerFramePair.right;
+      RexBuilder rexBuilder = relBuilder.getRexBuilder();
+
+      int groupSize = (int) corDefOutputs.keySet().stream()
+          .filter(a -> a.corr.equals(outerFramePair.left))
+          .count();
+      List<RelDataTypeField> newRelFields = newRel.getRowType().getFieldList();
+      ImmutableBitSet.Builder corFieldBuilder = ImmutableBitSet.builder();
+      final Map<Integer, RexNode> specificRexNodeMap = new HashMap<>();
```

Comment:
can you describe the keys and values of this map?
seems to map an index to a field access expression.